etc/greenboot/green.d/00_grub_fallback.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+set -euo pipefail
+
+# We need to determine whether this boot was a fallback boot
+if grub2-editenv list | grep -q "fallback_active=true"; then
+  grub2-editenv - unset fallback_active
+  grub2-editenv - set boot_success=0 # this should already be 0
+  grub2-editenv - set boot_counter=2
+  rpm-ostree rollback --reboot
```

Comment:
@jlebon, do we need to run `rollback` here? or just some form of `rpm-ostree cleanup` ?

Reply:
It depends on the behaviour we want I suppose. I'm slightly in favour of rollback to keep the bad deployment around in case admins want to prod it manually. A later upgrade will wipe it out. Though `cleanup` is fine too, assuming we at least write down information (e.g. version, checksum, timestamp) about the deployment that failed somewhere before removing it? Anyway, we'll likely need to revisit this once time comes to integrate this with rpm-ostree, where we'll want to make sure we don't try to auto-update to the same broken deployment again.

Either way, for now it seems like we want something like this?

```
rpm-ostree rollback
grub2-editenv - set boot_success=1
```

Then we don't even need to reset `boot_success` and `boot_counter`, right?